bundles/org.eclipse.cdt.lsp/src/org/eclipse/cdt/lsp/server/enable/HasLanguageServerPropertyTester.java

```diff
@@ -26,9 +38,49 @@ public HasLanguageServerPropertyTester() {
 	@Override
 	public boolean test(Object receiver, String property, Object[] args, Object expectedValue) {
 		if (cLanguageServerProvider != null) {
-			return cLanguageServerProvider.isEnabledFor(receiver);
+			if (receiver instanceof URI) {
+				// called from the language server enabler for LSP4E:
+				var uri = (URI) receiver;
+				var fileHandle = getFileHandle(uri);
+				if(fileHandle != null && fileHandle instanceof IResource) {
+					var resource = (IResource) fileHandle;
```

Comment:
Perhaps, too many `var` reywords. We have instance of `IResource` already. How could it became `null` at this point?